vpr/src/route/rr_graph.cpp

```diff
@@ -1367,9 +1367,9 @@ void free_rr_graph() {
 
     device_ctx.switch_fanin_remap.clear();
 
-    device_ctx.rr_node_metadata.clear();
+    device_ctx.rr_graph_builder.rr_node_metadata_clear();
```

Comment:
I suggest to

- Convert the ``rr_node_metadata_clear()`` and ``rr_edge_metadata_clear()`` function to private. 
- Call these clear functions in the ``clear()`` API of ``RRGraphBuilder``

Then, here you just need to call ``device_ctx.rr_graph_builder.clear()``

Reply:
I have called the rr_node_metadata_.clear() inside the RRGraphBuilder::clear. I have done this following the convention used for node_loopup_.clear() 
https://github.com/RapidSilicon/vtr-verilog-to-routing/blob/05dd5c00e92c2cd0ff958f750be9318a05973a2c/vpr/src/device/rr_graph_builder.cpp#L62-L66